typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -75,6 +76,24 @@ export abstract class HyperlaneDeployer<
   ) {
     this.logger = options?.logger ?? debug('hyperlane:deployer');
     this.chainTimeoutMs = options?.chainTimeoutMs ?? 5 * 60 * 1000; // 5 minute timeout per chain
+
+    if (this.options?.ismFactory) {
+      this.options.ismFactory.deployContractFromFactory =
+        this.deployContractFromFactory.bind(this);
+    }
```

Comment:
This is the first time I've seen a class method being passed around this way. My concern is that the ismFactory method, which is a separate class from this one, can modify this class' data and have unexpected side-effects. 

I don't understand well enough what problem we're solving here but I guess my first thought would be to use composition instead of this approach. I.e. pass the ism factory a reference to this deployer and have it call it as needed

Reply:
composing means exposing the deploy function, continued discussion here: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3255/commits/75781f5966ce6f94274e9225f2503d24ad699684#r1497906716